spotify_project/base/urls.py

```diff
@@ -9,4 +9,5 @@
     re_path(r'^callback/$', views.spotify_callback, name='callback'),
     re_path(r'^pull_data/$', views.pull_data, name='pull_data'),
     re_path(r'^all_artists/$', views.all_artists, name='all_artists'),
+    re_path(r'^single_artist/(?P<artist_id>[^/]+)/(?P<name>[^/]+)/$', views.single_artist, name='single_artist'),
```

Comment:
The `name` in here is not required. You only need the ID. Also, if you use `path` instead of `re_path` you can do something like: `single_artist/<str:artist_id>/`. Check out the docs, I haven't needed to do this in awhile. But it's way cleaner to read than using a regex.